Entity/Components/EnergyComponent.gd

```diff
@@ -0,0 +1,38 @@
+extends EntityComponent
+class_name EnergyComponent
+
+@export var max_energy: int = 4
+@export var starting_energy: int = 0
+@export var energy_generation: int = 3
+@export var ignore_cost: bool = false
+var energy = starting_energy
+
+signal on_energy_changed(new_energy: int)
+
+func is_playable(card: CardWorld) -> bool:
+	if ignore_cost:
+		return true
+
+	if card != null:
+		return energy >= card.card_data.energy_cost
+
+	return true
+
+
+func use_energy(card: CardBase) -> void:
+	energy -= card.energy_cost
+
+	if energy < 0:
```

Comment:
this might hide a problem (energy shouldn't be negative)
instead of setting the energy back to 0, we should assert the energy is not lower than 0 instead